cmd/ipfs/main.go

```diff
@@ -300,65 +299,113 @@ func callPreCommandHooks(ctx context.Context, details cmdDetails, req cmds.Reque
 	return nil
 }
 
-func callCommand(ctx context.Context, req cmds.Request, root *cmds.Command, cmd *cmds.Command) (cmds.Response, error) {
+func callCommand(ctx context.Context, req cmds.Request, root *cmds.Command, cmd *cmds.Command) error {
 	log.Info(config.EnvDir, " ", req.InvocContext().ConfigRoot)
-	var res cmds.Response
 
 	err := req.SetRootContext(ctx)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	details, err := commandDetails(req.Path(), root)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	client, err := commandShouldRunOnDaemon(*details, req, root)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	err = callPreCommandHooks(ctx, *details, req, root)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
+	encTypeStr, found, err := req.Option("encoding").String()
+	if !found || err != nil {
+		//encTypeStr = cmds.Text
+		panic(err)
+	}
+	encType := cmds.EncodingType(encTypeStr)
+
+	log.Debug("creating RE with encType ", encType)
+
+	var (
+		re     cmds.ResponseEmitter
+		exitCh <-chan int
+	)
+
+	if enc, ok := cmd.Encoders[encType]; ok {
+		re, exitCh = cli.NewResponseEmitter(os.Stdout, os.Stderr, enc, req)
+	} else if enc, ok := cmds.Encoders[encType]; ok {
+		re, exitCh = cli.NewResponseEmitter(os.Stdout, os.Stderr, enc, req)
```

Comment:
actually one of them is cmd.Encoders and the other is cmd**s**.Encoders. I'll put a comment on top that makes this clear.